imports/lib/schemas/ChatMessage.ts

```diff
@@ -1,29 +1,27 @@
-import * as t from 'io-ts';
-import { date } from 'io-ts-types';
-import { BaseCodec, BaseOverrides } from './Base';
+import { z } from 'zod';
+import { nonEmptyString } from './customTypes';
 import { Id } from './regexes';
-import type { Overrides } from './typedSchemas';
-import { buildSchema, inheritSchema } from './typedSchemas';
+import withCommon from './withCommon';
 
-const MentionBlock = t.type({
-  type: t.literal('mention'),
-  userId: t.string,
+const MentionBlock = z.object({
+  type: z.literal('mention'),
+  userId: z.string().regex(Id),
 });
-export type ChatMessageMentionNodeType = t.TypeOf<typeof MentionBlock>;
+export type ChatMessageMentionNodeType = z.infer<typeof MentionBlock>;
 
-const TextBlock = t.type({
-  text: t.string,
+const TextBlock = z.object({
+  text: nonEmptyString,
```

Comment:
Due to some constraints from slate (and a desire to preserve the validity of `ChatMessageContent` as a slate document) the `text` here may actually be an empty string, and should be stored and retrieved as such.

Reply:
Ah, fair. I had forgotten about that quirk of slate documents. I've introduced an `allowedEmptyString` that can be used for this sort of thing.